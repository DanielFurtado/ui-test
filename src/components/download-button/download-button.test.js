import { render, fireEvent } from '@testing-library/react';
import renderer from 'react-test-renderer';
import DownloadButton from './download-button.component';

it('renders', () => {
  const component = renderer.create(<DownloadButton />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it('should call window open with the correct url', () => {
  window.open = jest.fn();
  const url = 'https://www.some-link.com/';
  const { getByText } = render(<DownloadButton url={url} />);
  const downloadButton = getByText('Download');
  
  fireEvent.click(downloadButton);

  expect(window.open).toHaveBeenCalledTimes(1);
  expect(window.open).toHaveBeenCalledWith('https://www.some-link.com/', '_blank');
})
