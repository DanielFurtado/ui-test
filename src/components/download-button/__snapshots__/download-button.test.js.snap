// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<button
  aria-label="Download"
  className="button-primary"
  onClick={[Function]}
>
  Download
</button>
`;
